### PR TITLE
Updated Github Actions Workflows

### PR DESCRIPTION
**Pull request by steve-the-edwards**
Adding a runner to run ios KMP tests, and just went a bit wild updating everything.

I cribbed the ios runner recipe from sqldelight, as you do.

### CODE REVIEW

File: .github/workflows/kotlin.yml [hunk 1]
```diff
@@ -25,10 +25,11 @@ jobs :
...
     steps :
-      - uses : actions/checkout@v2
-      - name : set up JDK 11.0.7
-        uses : actions/setup-java@v1
+      - uses : actions/checkout@v3
+      - name : set up JDK 11
+        uses : actions/setup-java@v3
         with :
+          distribution: 'zulu'
           java-version : 11.0.7
```
Comment (RBusarow) [inline .github/workflows/kotlin.yml:33]:
```suggestion
          java-version : 11
```

File: .github/workflows/kotlin.yml [hunk 2]
```diff
@@ -64,11 +65,12 @@ jobs :
...
-      - uses : actions/checkout@v2
+      - uses : actions/checkout@v3
       - uses : gradle/wrapper-validation-action@v1
-      - name : set up JDK 11.0.7
-        uses : actions/setup-java@v1
+      - name : set up JDK 11
+        uses : actions/setup-java@v3
         with :
+          distribution: 'zulu'
           java-version : 11.0.7
```
Comment (RBusarow) [inline .github/workflows/kotlin.yml:74]:
```suggestion
          java-version : 11
```

**Review by RBusarow — APPROVED** [review]
Nice!

File: .github/workflows/kotlin.yml (hunk 1 above)
Comment (steve-the-edwards) [reply .github/workflows/kotlin.yml:33]: Whoops. Missed these.

**Comment by bnvinay92** [comment]
Thanks!

**Pull request merged**